yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtArray.kt

```diff
@@ -1,3 +1,162 @@
 package dev.yorkie.document.crdt
 
-public class CrdtArray
+import dev.yorkie.document.time.TimeTicket
+
+/**
+ * [CrdtArray] represents Array data type containing logical clocks.
+ */
+internal class CrdtArray(
+    val elements: RgaTreeList,
+    createdAt: TimeTicket,
+) : CrdtContainer(createdAt), Iterable<CrdtElement> {
+    val head
+        get() = elements.dummyHead.value
+
+    val last
+        get() = elements.last.value
+
+    val length
+        get() = elements.length
+
+    /**
+     * Returns the sub path of the given [createdAt] element.
+     */
+    override fun subPathOf(createdAt: TimeTicket): String? {
+        return elements.subPathOf(createdAt)
+    }
+
+    // NOTE(7hong13): Original comment from JS SDK is as follows:
+    // `purge` physically purge child element.
+    /**
+     * Physically purges the given [element].
+     */
+    override fun purge(element: CrdtElement) {
+        elements.purge(element)
+    }
+
+    // NOTE(7hong13): Original comment from JS SDK is as follows:
+    // `insertAfter` adds a new node with the value after the given node.
+    /**
+     * Adds a new node with [value] after the node created at [prevCreatedAt].
+     */
+    fun insertAfter(prevCreatedAt: TimeTicket, value: CrdtElement) {
+        elements.insertAfter(prevCreatedAt, value)
+    }
+
+    /**
+     * Moves the given [createdAt] element after the [prevCreatedAt] element.
+     */
+    fun moveAfter(prevCreatedAt: TimeTicket, createdAt: TimeTicket, executedAt: TimeTicket) {
+        elements.moveAfter(prevCreatedAt, createdAt, executedAt)
+    }
+
+    /**
+     * Returns the element of the given [createdAt].
+     */
+    fun get(createdAt: TimeTicket): CrdtElement? {
+        val node = elements.get(createdAt)
+        if (node?.isRemoved == true) return null
+        return node
+    }
+
+    /**
+     * Returns the element of the given [index].
+     */
+    fun getByIndex(index: Int): CrdtElement? {
```

Comment:
maybe we could make this `operator fun`

Reply:
@hackerwins we can use `operator fun` in kotlin for some benefits, but then we need to rename this function to `get`, instead of `getByIndex`.
is it ok to have different function name with js-sdk?

Reply:
Never know about `operator fun` before. Can we use `operator fun` for `TimeTicket` and `Index` at the same time(method overloading)?

Reply:
@hackerwins yes, it seems like method overloading is allowed here (I see no compile time error)